twenty-first/src/math/bfield_codec.rs

```diff
@@ -648,6 +690,116 @@ mod tests {
     neg_test_case! { fn vec_of_vec_of_bfield_elements_neg for Vec<Vec<BFieldElement>> }
     neg_test_case! { fn vec_of_vec_of_xfield_elements_neg for Vec<Vec<XFieldElement>> }
 
+    #[test]
+    fn trailing_zeros_have_no_effect_on_encoding_empty_poly_bfe() {
+        let poly_no_trailing_zeros: Polynomial<BFieldElement> = Polynomial::new(vec![]);
+        let encoded = poly_no_trailing_zeros.encode();
+
+        let mut poly_w_trailing_zeros: Polynomial<BFieldElement> = Polynomial::new(vec![]);
+        poly_w_trailing_zeros.coefficients.push(bfe!(0));
+        let encoded_of_trailing_zeros_poly = poly_w_trailing_zeros.encode();
+
+        assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[test]
+    fn trailing_zeros_have_no_effect_on_encoding_empty_poly_xfe() {
+        let poly_no_trailing_zeros: Polynomial<XFieldElement> = Polynomial::new(vec![]);
+        let encoded = poly_no_trailing_zeros.encode();
+
+        let mut poly_w_trailing_zeros: Polynomial<XFieldElement> = Polynomial::new(vec![]);
+        poly_w_trailing_zeros.coefficients.push(xfe!(0));
+        let encoded_of_trailing_zeros_poly = poly_w_trailing_zeros.encode();
+
+        assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn trailing_zeros_have_no_effect_on_encoding_poly_bfe_pbt(
+        mut polynomial: Polynomial<BFieldElement>,
+        #[strategy(0usize..30)] num_leading_zeros: usize,
+    ) {
+        let encoded = polynomial.encode();
+        polynomial
+            .coefficients
+            .extend(vec![BFieldElement::zero(); num_leading_zeros]);
+        let encoded_of_trailing_zeros_poly = polynomial.encode();
+        prop_assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn trailing_zeros_have_no_effect_on_encoding_poly_xfe_pbt(
+        mut polynomial: Polynomial<XFieldElement>,
+        #[strategy(0usize..30)] num_leading_zeros: usize,
+    ) {
+        let encoded = polynomial.encode();
+        polynomial
+            .coefficients
+            .extend(vec![XFieldElement::zero(); num_leading_zeros]);
+        let encoded_of_trailing_zeros_poly = polynomial.encode();
+        prop_assert_eq!(encoded, encoded_of_trailing_zeros_poly);
+    }
+
+    #[proptest]
+    fn disallow_trailing_zeros_in_poly_encoding_bfe(
+        mut polynomial: Polynomial<BFieldElement>,
+        #[filter(!#leading_coefficient.is_zero())] leading_coefficient: BFieldElement,
+        #[strategy(1usize..30)] num_leading_zeros: usize,
```

Comment:
Mix of lingo: are the zeros trailing or leading? I'm aware that both perspectives are valid and would argue for consistency.

Reply:
The encoding can have trailing zeros. As long as you specify that you're talking about the *encoding* I think it's unambiguous but I agree that there is potential for confusion.